pmlc/rt/device_id.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Intel Corporation
+
+#include "pmlc/rt/device_id.h"
+
+#include "llvm/Support/FormatVariadic.h"
+
+#include "pmlc/rt/internal.h"
+#include "pmlc/rt/runtime_registry.h"
+#include "pmlc/util/logging.h"
+
+namespace pmlc::rt {
+
+const std::regex &getDeviceIDRegex() {
+  static const std::regex re(
+      "([[:alpha:]][_[:alnum:]]*(-[[:alpha:]][_[:alnum:]]*)*)[.]([[:digit:]]+)",
```

Comment:
Can we just split on `.` to avoid having to use a regex?

Reply:
Sure -- just curious, what's the rationale?  (I liked that this made the format for device IDs clear, and that it didn't require any subsequent validation on the ordinal.  NB the device ID -> device translation only occurs when the `Executable` is created; this isn't something that happens for every inference (although I doubt it'd noticeable if it did).

Reply:
Simplicity